### PR TITLE
Refactor GitHub Actions, add Coding Standards to the CI pipeline.

### PR DESCRIPTION
**Pull request by sidz**
This PR:
- [x] Changes GitHub Actions flow
- [x] Adds Coding Standard checks to the CI pipeline.
- [x] remove Macos as it doesn't make sense and waste of resources and time.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,29 +1,66 @@
...
     strategy:
       matrix:
-        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
-        php-version: ['7.1', '7.2', '7.3', '7.4', 8.0']
-    name: PHP ${{ matrix.php-version }} Test on ${{ matrix.operating-system }}
+        operating-system: [ ubuntu-latest ]
+        php-version: ['7.4']
+        dependencies: ['']
+        include:
+          - { operating-system: 'ubuntu-latest', php-version: '7.1', dependencies: '--prefer-lowest' }
```
Comment (exussum12) [inline .github/workflows/ci.yaml:22]: This is testing way less than what is there now, mac is missing and windows only has 1 version. We could double the existing tests with the prefer lowest flag ?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # PHP Magic Number Detector (PHPMND)
 
+[![Minimum PHP version: 7.1.0](https://img.shields.io/badge/php-7.1.0%2B-blue.svg)](https://packagist.org/packages/povils/phpmnd)
```
Comment (exussum12) [inline README.md:3]: +1 Great addition

File: tests/Printer/XmlTest.php [hunk 3]
```diff
@@ -66,7 +69,9 @@ public function testPrintData() : void
     <files>
         <file errors="1" path="Foo/Bar.php">
             <entry end="82" line="1" start="80">
-                <snippet><![CDATA[$rootNode->setAttribute('fileCount', count($fileReportList->getFileReports()) + 12);]]></snippet>
+                <snippet>
+                    <![CDATA[$rootNode->setAttribute('fileCount', count($fileReportList->getFileReports()) + 12);]]>
```
Comment (exussum12) [inline tests/Printer/XmlTest.php:73]: This 12 was removed wasn't it ?

**Review by exussum12 — APPROVED** [review]

File: tests/Printer/XmlTest.php (hunk 3 above)
Comment (sidz) [reply tests/Printer/XmlTest.php:73]: as you can see :) I just fixed XML to follow our standards.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (sidz) [reply .github/workflows/ci.yaml:22]: IMHO we do not need to test on Mac as it waste of time. Testing on linux on 100% covers Mac.
Also I've never seen that anybody in php community doing this. 

we just need to check that application can is working  find on windows. All other cases will be covered on linux.

**Comment by sidz** [comment]
merged. thanks @exussum12

**Pull request merged**